### PR TITLE
[DEV-1119] add profile personal data page

### PR DESCRIPTION
**Pull request by MarcoPonchia**
#### List of Changes

Add personal data page

#### Motivation and Context

Logged user can see registration's personal information

#### How Has This Been Tested?

After login user can naviagate on:  http://localhost:3000/profile/personal-data

#### Screenshots (if appropriate):
screenshot-localhost_3000-2023 11 20-14_04_56

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/app/profile/personal-data/page.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+'use client';
+
+import { Stack, Typography } from '@mui/material';
+import { useTranslations } from 'next-intl';
+import { useUser } from '@/helpers/user.helper';
+import { InfoCardItemProps } from '@/components/atoms/InfoCardItem/InfoCardItem';
+import { InfoCard } from '@/components/molecules/InfoCard/InfoCard';
+import PageNotFound from '@/app/not-found';
+
+const TermsOfService = () => {
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/profile/personal-data/page.tsx:10]: Why "TermsOfService" and not "PersonalData"?

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/messages/it.json [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
     "shared": {
-        "goToModel": "Vai al modello"
+        "goToModel": "Vai al modello",
+        "edit": "Modifica"
+    },
+    "profile": {
+        "personalData": {
+            "title": "I tuoi dati",
+            "dataSection": "Profilo",
+            "accountSection": "Profilo",
```
Comment (kin0992) [inline apps/nextjs-website/src/messages/it.json:10]:
```suggestion
            "accountSection": "Account",
```

File: apps/nextjs-website/src/messages/it.json [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+            "fields": {
+                "name": "Nome",
+                "surname": "Cognome",
+                "role": "Ruolo",
+                "company": "Tipologia di ente o azienda",
+                "sector": "Settore",
+                "products": "Prodotti di interesse",
+                "email": "Indirizzo e-mail",
+                "password": "password"
```
Comment (kin0992) [inline apps/nextjs-website/src/messages/it.json:19]:
```suggestion
                "password": "Password"
```

File: apps/nextjs-website/src/app/profile/personal-data/page.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  ];
+
+  // TODO: add dedicated loading and unauthorized pages
+  if (!loading && !user) {
+    return <PageNotFound />;
+  }
+
+  return (
+    <Stack gap={5} sx={{ padding: 5, width: '100%' }}>
+      <Typography variant='h3'>{t('personalData.title')}</Typography>
```
Comment (kin0992) [inline apps/nextjs-website/src/app/profile/personal-data/page.tsx:59]:
```suggestion
      <Typography variant='h4'>{t('personalData.title')}</Typography>
```

File: apps/nextjs-website/src/components/atoms/InfoCardItem/InfoCardItem.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+export type InfoCardItemProps = {
+  title: string;
+  value?: string;
+};
+
+export const InfoCardItem = ({ title, value }: InfoCardItemProps) => {
+  return (
+    <Stack my={{ xs: 1, md: 3 }} flexDirection={{ xs: 'column', md: 'row' }}>
+      <Typography fontSize={16} minWidth={{ xs: 'auto', md: '200px' }}>
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/InfoCardItem/InfoCardItem.tsx:13]: I think this should be a `body2` variant

File: apps/nextjs-website/src/app/profile/personal-data/page.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      // TODO: add company when available
+    },
+    {
+      title: t('personalData.fields.sector'),
+      value: user?.attributes['custom:company_type'],
+    },
+    {
+      title: t('personalData.fields.products'),
+      // TODO: add products when available
+    },
```
Comment (kin0992) [inline apps/nextjs-website/src/app/profile/personal-data/page.tsx:38]:
```suggestion
```
At the moment I'd remove this. We don't even know yet how (nor when) the user could populate that information

File: apps/nextjs-website/src/app/profile/personal-data/page.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      value: user?.attributes.family_name,
+    },
+    {
+      title: t('personalData.fields.role'),
+      value: user?.attributes['custom:job_role'],
+    },
+    {
+      title: t('personalData.fields.company'),
+      // TODO: add company when available
+    },
```
Comment (kin0992) [inline apps/nextjs-website/src/app/profile/personal-data/page.tsx:30]:
```suggestion
```
same as another comment

**Review by kin0992 — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/atoms/InfoCardEditButton/InfoCardEditButton.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import { useTranslations } from 'next-intl';
+import EditIcon from '@mui/icons-material/Edit';
+
+const InfoCardEditButton = ({ onClick }: { onClick?: () => null }) => {
+  const t = useTranslations('shared');
+
+  return (
+    <Button
+      onClick={onClick}
+      sx={{ paddingLeft: 0, paddingRight: 0 }}
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/InfoCardEditButton/InfoCardEditButton.tsx:12]: When the user clicks on the button, the lack of padding is a problem:
image.
I think you should use the ButtonNaked component

File: apps/nextjs-website/src/components/molecules/InfoCard/InfoCard.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  cardTitle: string;
+  items: InfoCardItemProps[];
+};
+
+export const InfoCard = ({ cardTitle, items }: InfoCardProps) => {
+  const editButton = !isProduction ? <InfoCardEditButton /> : null;
+  return (
+    <Card raised sx={{ padding: 4, maxWidth: '700px' }}>
+      <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
+        <Typography variant='h6' mt={1}>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/molecules/InfoCard/InfoCard.tsx:20]: The fontWeight should be `700`
image

File: apps/nextjs-website/src/app/profile/personal-data/page.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+      <InfoCard
+        cardTitle={t('personalData.dataSection')}
+        items={dataSectionItems}
+      />
+
+      <InfoCard
+        cardTitle={t('personalData.accountSection')}
+        items={accountSectionItems}
+      />
+    </Stack>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/profile/personal-data/page.tsx:61]: You should add the "delete user" section (see this [link](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_DeleteUser.html) for the doc)

Comment (MarcoPonchia) [reply apps/nextjs-website/src/app/profile/personal-data/page.tsx:60]: The deleted section is realtive to a different [task](https://pagopa.atlassian.net/jira/software/projects/DEV/boards/280?selectedIssue=DEV-1020) and will be added in a separte PR.

File: apps/nextjs-website/src/components/atoms/InfoCardItem/InfoCardItem.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        {title}
+      </Typography>
+      <Typography
+        minHeight={'24px'}
+        fontSize={16}
+        flexGrow={1}
+        fontWeight={700}
+      >
+        {value}
+      </Typography>
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/InfoCardItem/InfoCardItem.tsx:27]: `value` is optional: if the component receives an `undefined` or `null` value, what is going to be rendered?

**Review by kin0992 — APPROVED** [review]

**Pull request merged**